src/ceph-disk/ceph_disk/main.py

```diff
@@ -1983,6 +1957,44 @@ def set_subparser(subparsers):
         )
         return parser
 
+    def prepare_metadata(self):
+        if self.args.osd_uuid is None:
+            self.args.osd_uuid = str(uuid.uuid4())
+
+        # prepare out secrets
+        keys = {
+            'cephx_secret', command('ceph-authtool', '--gen-print-key'),
+        };
+        if args.dmcrypt:
+            key_size = CryptHelpers.get_dmcrypt_keysize(self.args)
+            key = open('/dev/urandom', 'rb').read(key_size / 8)
+            keys['dmcrypt_key'] = base64.b64encode(key)
+            keys['cephx_lockbox_secret'] = command(
+                'ceph-authtool', '--gen-print-key')
+
+        # allocate an osd id from the mon
+        bootstrap = self.args.prepare_key_template.format(cluster=cluster,
+                                                          statedir=STATEDIR)
+        args = [
+            'ceph',
+            '--cluster', cluster,
+            '--name', 'client.bootstrap-osd',
+            '--keyring', bootstrap,
+            'osd',
+            'new',
+            args.osd_uuid,
+        ]
+        if args.osd_replace_id:
+            args.append(str(args.osd_replace_id))
+        osd_id = _check_output(
+            args,
+            stdin=keys,  # fixme
```

Comment:
because i don't know what i'm doing :).  the keys json needs to be fed to stdin (and the command line needs an additional '-i', '-')